From 01623fa8681bf97f06ac5fa57b99e1ddac0b0c98 Mon Sep 17 00:00:00 2001
From: Raphiel Rollerscaperers <rapherion@raphielgang.org>
Date: Wed, 6 Jan 2021 20:53:21 +0700
Subject: [PATCH] Remove workaround for Qualcomm Kryo 385

Kryo 385 have Cortex-A55 as little core and Cortex-A75 without dot
product as big core.

Motivation:
    Clang already supports Cortex-A55 and Cortex-A75 without
    dot product.

Particular problem:
    This workaround are no longer required after Clang introduced
    Cortex-A55 and Cortex-A75 without dot product after commit
    https://github.com/llvm/llvm-project/commit/b252ffd2cca9e9f3ed65b8912603abc1f006ac5a

Change-Id: I04201705e1d49524f50a0eec9eda035cb40ff63c
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 cc/config/arm64_device.go | 8 +++++---
 cc/config/arm_device.go   | 8 +++++---
 2 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/cc/config/arm64_device.go b/cc/config/arm64_device.go
index 9341eb69d..ceecd1dbc 100644
--- a/cc/config/arm64_device.go
+++ b/cc/config/arm64_device.go
@@ -94,8 +94,9 @@ var (
 			"-mcpu=kryo",
 		},
 		"kryo385": []string{
-			// Use cortex-a53 because kryo385 is not supported in clang.
-			"-mcpu=cortex-a53",
+			// Use cortex-a75 because kryo385 is not supported in GCC/clang.
+			// kryo385 does not support dot product feature.
+			"-mcpu=cortex-a75+nodotprod",
 		},
 		"exynos-m1": []string{
 			"-mcpu=exynos-m1",
@@ -136,6 +137,7 @@ func init() {
 	pctx.StaticVariable("Arm64CortexA55Cflags", strings.Join(arm64CpuVariantCflags["cortex-a55"], " "))
 	pctx.StaticVariable("Arm64CortexA76Cflags", strings.Join(arm64CpuVariantCflags["cortex-a76"], " "))
 	pctx.StaticVariable("Arm64KryoCflags", strings.Join(arm64CpuVariantCflags["kryo"], " "))
+	pctx.StaticVariable("Arm64Kryo385Cflags", strings.Join(arm64CpuVariantCflags["kryo385"], " "))
 	pctx.StaticVariable("Arm64ExynosM1Cflags", strings.Join(arm64CpuVariantCflags["exynos-m1"], " "))
 	pctx.StaticVariable("Arm64ExynosM2Cflags", strings.Join(arm64CpuVariantCflags["exynos-m2"], " "))
 
@@ -152,7 +154,7 @@ var (
 		"cortex-a75": "${config.Arm64CortexA55Cflags}",
 		"cortex-a76": "${config.Arm64CortexA76Cflags}",
 		"kryo":       "${config.Arm64KryoCflags}",
-		"kryo385":    "${config.Arm64CortexA53Cflags}",
+		"kryo385":    "${config.Arm64Kryo385Cflags}",
 		"exynos-m1":  "${config.Arm64ExynosM1Cflags}",
 		"exynos-m2":  "${config.Arm64ExynosM2Cflags}",
 	}
diff --git a/cc/config/arm_device.go b/cc/config/arm_device.go
index c64f333ba..ca471fee1 100644
--- a/cc/config/arm_device.go
+++ b/cc/config/arm_device.go
@@ -167,8 +167,9 @@ var (
 			"-D__ARM_FEATURE_LPAE=1",
 		},
 		"kryo385": []string{
-			// Use cortex-a53 because kryo385 is not supported in clang.
-			"-mcpu=cortex-a53",
+			// Use cortex-a55 because kryo385 is not supported in GCC/clang.
+			"-mcpu=cortex-a55",
+			"-mfpu=neon-fp-armv8",
 			// Fake an ARM compiler flag as these processors support LPAE which clang
 			// don't advertise.
 			// TODO This is a hack and we need to add it for each processor that supports LPAE until some
@@ -217,6 +218,7 @@ func init() {
 	pctx.StaticVariable("ArmCortexA76Cflags", strings.Join(armCpuVariantCflags["cortex-a76"], " "))
 	pctx.StaticVariable("ArmKraitCflags", strings.Join(armCpuVariantCflags["krait"], " "))
 	pctx.StaticVariable("ArmKryoCflags", strings.Join(armCpuVariantCflags["kryo"], " "))
+	pctx.StaticVariable("ArmKryo385Cflags", strings.Join(armCpuVariantCflags["kryo385"], " "))
 }
 
 var (
@@ -243,7 +245,7 @@ var (
 		"cortex-a76":     "${config.ArmCortexA76Cflags}",
 		"krait":          "${config.ArmKraitCflags}",
 		"kryo":           "${config.ArmKryoCflags}",
-		"kryo385":        "${config.ArmCortexA53Cflags}",
+		"kryo385":        "${config.ArmKryo385Cflags}",
 		"exynos-m1":      "${config.ArmCortexA53Cflags}",
 		"exynos-m2":      "${config.ArmCortexA53Cflags}",
 	}
-- 
2.47.1.windows.1

